frontend/src/components/dashboard/gallery-components/MyArtsGallery.js

```diff
@@ -0,0 +1,53 @@
+import React, {useEffect} from 'react';
+import Grid from '@material-ui/core/Grid';
+import ImgMediaCard from './ImgMediaCard';
+import {generateTextToSpeech,
+  getUserAudioTranscript} from '../textToSpeechHelpers';
+import {useSelector} from 'react-redux';
+
+
+export default function MyArtsGallery() {
+  const artworksMap = useSelector((state) =>
+    state.myArtworks.artworks);
+  const loading = useSelector((state) =>
+    state.myArtworks.loading);
+  const artworks = Array.from(artworksMap);
+
+  let cards;
+  if (!loading && artworks) {
+    cards = artworks.map(([key, artwork]) => (
+      <Grid key={key} item>
+        <ImgMediaCard
+          name={artwork.get('title')}
+          link={`/gallery/user/${key}`}
+          alt={artwork.get('alt')}
+          url={artwork.get('url')}
+        />
+      </Grid>
+    ));
+  }
+
+  // generate audio before users view artwork to reduce audio wait time
+  useEffect(() => {
+    if (!loading && artworks) {
+      artworks.forEach(([id, artwork]) => {
+        generateTextToSpeech(
+            getUserAudioTranscript(artwork), id);
+      });
+    }
+  });
+
+  return (
+    <>
+      <Grid
+        container
+        direction="row"
+        justify="center"
+        alignItems="center"
+        spacing={4}
+      >
+        {artworks ? cards : <p>Loading...</p>}
+      </Grid>
+    </>
+  );
+}
```

Comment:
I feel like we could restructure Gallery component so that you can use it for "My Artwork Gallery" too but we would have to do some refactoring since a lot of it is based off of whether the gallery isMuseum or not.  In hindsight we should have made the Gallery component more versatile so it could be used here.  Not sure if it's worth it to refactor everything or just have this separate component?? What do you guys think?

Reply:
I think this can use the Gallery component since these will all be user gallery artworks. Though, with the favorites I might need to do something like this because the artworks array would have whether its museum or user unlike everywhere else. Could refactor to make them include it at the artwork level so that gallery would no longer have that prop.

Reply:
After reevaluating don't think it's worth and separate components are fine.